pkg/frontend/read_path/query_service_handler.go

```diff
@@ -203,10 +204,16 @@ func (r *Router) GetProfileStats(
 
 func (r *Router) ProfileTypes(
 	ctx context.Context,
-	req *connect.Request[querierv1.ProfileTypesRequest],
+	c *connect.Request[querierv1.ProfileTypesRequest],
 ) (*connect.Response[querierv1.ProfileTypesResponse], error) {
-	if r.frontend != nil {
-		return r.frontend.ProfileTypes(ctx, req)
-	}
-	return connect.NewResponse(&querierv1.ProfileTypesResponse{}), nil
+	return Query[querierv1.ProfileTypesRequest, querierv1.ProfileTypesResponse](ctx, r, c,
+		func(a, b *querierv1.ProfileTypesResponse) (*querierv1.ProfileTypesResponse, error) {
```

Comment:
this merges data coming from the v1 and the v2 read paths, when the time range includes the transition between the two